lldb/source/Target/Process.cpp

```diff
@@ -3265,18 +3275,25 @@ Status Process::PrivateResume() {
     // (suspended/running/stepping). Threads should also check their resume
     // signal in lldb::Thread::GetResumeSignal() to see if they are supposed to
     // start back up with a signal.
-    if (m_thread_list.WillResume()) {
+    RunDirection direction;
```

Comment:
I think it would be a lot cleaner to have Process::CanReverseContinue, and check that before WillResume or anything.  If you are asked to go in reverse and it's not supported, we should do as little work as possible before returning an error.

Reply:
OK, let me gently disagree about this :-). In terms of code complexity, adding a call to `Process::CanReverseContinue()` is a tiny bit more complex than not having it. In terms of performance, I don't think it's important to make it a tiny bit faster to report that reverse-continue is not supported. Is there some other consideration?